modular_skyrat/modules/customization/modules/client/preferences.dm

```diff
@@ -3372,8 +3372,10 @@ GLOBAL_LIST_INIT(food, list(
 
 /datum/preferences/proc/get_linguistic_points()
 	var/points
-	points = LINGUISTIC_POINTS_DEFAULT
-	points = (TRAIT_LINGUIST in all_quirks) ? points + LINGUISTIC_POINTS_LINGUIST : points
+	if("Linguist" in all_quirks)
+		points = LINGUISTIC_POINTS_LINGUIST
+	else
+		points = LINGUISTIC_POINTS_DEFAULT
```

Comment:
Instead of all this just use
`points = (TRAIT_LINGUIST in all_quirks) ? LINGUISTIC_POINTS_LINGUIST : LINGUISTIC_POINTS_DEFAULT`